src/test/java/com/revlearn/team1/unit/controller/CourseControllerTest2.java

```diff
@@ -0,0 +1,193 @@
+package com.revlearn.team1.unit.controller;
+
+import com.revlearn.team1.controller.CourseController;
+import com.revlearn.team1.dto.CourseDTO;
+import com.revlearn.team1.dto.request.CourseEducatorDTO;
+import com.revlearn.team1.dto.response.CourseEducatorResDTO;
+import com.revlearn.team1.enums.AttendanceMethod;
+import com.revlearn.team1.service.CourseServiceImp;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@ExtendWith(MockitoExtension.class)
```

Comment:
why are there 2 test files? not an issue just curious

Reply:
I think the second one is the correct way to test a controller class, but I didn't understand that until I'd already implemented the first way.  Now it seems like a waste to delete.